state/accountsDB.go

```diff
@@ -1010,9 +1010,12 @@ func (adb *AccountsDB) recreateTrie(options common.RootHashHolder) error {
 
 // RecreateAllTries recreates all the tries from the accounts DB
 func (adb *AccountsDB) RecreateAllTries(rootHash []byte) (map[string]common.Trie, error) {
```

Comment:
The `SnapshotState` and `SetStateCheckpoint`  func uses the same pattern with leaves chan and err chan. Please use the newly created stuct instead.